src/plugins/intel_gpu/tests/unit/test_cases/loop_gpu_test.cpp

```diff
@@ -428,3 +429,121 @@ TEST(loop_gpu, basic_concat_cached) {
 TEST(loop_gpu, basic_concat_nested_cached) {
     test_loop_gpu_basic_concat_nested<float>(true);
 }
+
+static void test_loop_gpu_wo_trip_count(bool is_caching_test) {
+    auto& engine = get_test_engine();
+
+    auto e_input_layout = cldnn::layout{ { 1, 1, 5, 4 }, data_types::f32, format::bfyx };
+    auto b_input_layout = cldnn::layout{ { 1, 1, 1, 4}, data_types::f32, format::bfyx };
+    auto const_layout = cldnn::layout{ {}, data_types::i64, format::bfyx };
+
+    auto e_input_mem = engine.allocate_memory(e_input_layout); // b,f,x,y
+    auto e_initial_condition_mem = engine.allocate_memory(const_layout);
+    auto e_num_iteration_mem = engine.allocate_memory(const_layout);
+    auto b_exit_value_mem = engine.allocate_memory(const_layout);
+    auto b_index_inc_mem = engine.allocate_memory(const_layout);
+
+    std::vector<float> input_data{
+        1.0f,  2.0f, -15.f,  3.0f,
+        4.0f, -15.f, 5.0f,  6.0f,
+        -15.f, 7.0f, -15.f, 0.0f,
+        0.0f, -15.f, 0.5f, -0.5f,
+        -15.f, 8.0f,  1.5f,  5.2f
+    };
+
+    const int64_t exit_value = 3;
+
+    // initialize input buffers
+    set_values(e_input_mem, input_data);
+    set_values(e_initial_condition_mem, {1});
+    set_values(b_exit_value_mem, {exit_value});
+    set_values(b_index_inc_mem, {1});
+
+    primitive_id body_current_iteration_id = "b_index";
+    primitive_id body_execution_condition_id = "b_cond_exit_value";
+
+    cldnn::topology body(
+        input_layout(body_current_iteration_id, const_layout),
+        input_layout("b_add_data", b_input_layout),
+        input_layout("b_mul_data", b_input_layout),
+        data("b_exit_value", b_exit_value_mem),
+        data("b_index_inc", b_index_inc_mem),
+        eltwise("b_index_update", input_info(body_current_iteration_id), input_info("b_index_inc"), eltwise_mode::sum),
+        reorder("b_index_cast", input_info("b_index_update"),
+                    cldnn::format::any, data_types::f32, {}, cldnn::reorder_mean_mode::subtract, cldnn::padding(), true),
+        eltwise(body_execution_condition_id, input_info("b_index"), input_info("b_exit_value"), eltwise_mode::lt),
+        eltwise("b_add", input_info("b_add_data"), input_info("b_index_cast"), eltwise_mode::sum),
+        eltwise("b_mul", input_info("b_mul_data"), input_info("b_index_cast"), eltwise_mode::prod)
+    );
+
+    primitive_id trip_count_id = "";
+    primitive_id num_iteration_id = "num_iteration";
+    primitive_id initial_condition_id = "initial_condition";
+    int64_t num_iterations = -1;
+
+    std::vector<loop::io_primitive_map> input_primitive_maps {
+        loop::io_primitive_map("input", "b_add_data", 2),
+        loop::io_primitive_map("input", "b_mul_data", 2),
+        loop::io_primitive_map(num_iteration_id, body_current_iteration_id) };
```

Comment:
I am little bit confused, as I understand, "num_itearation" is the iteration count of actual iteration happened at a point. However, it is also being used as "max limit", i.e., -1. Am I understanding correctly? It sounds like num_iteration acts as both  "limitation" and "actually executed count".

Reply:
Yes, actually executed count is right. I think the name of variable makes you confused. I will change it. thaks.